packages/devtools_app/test/shared/extent_delegate_list_view_test.dart

```diff
@@ -80,5 +80,157 @@ void main() {
       );
       expect(pointerSignalEventCount, equals(1));
     });
+
+    testWidgets('inherits PrimaryScrollController automatically',
+        (tester) async {
+      final ScrollController controller = ScrollController();
+      await pumpList(
+        tester,
+        PrimaryScrollController(
+          controller: controller,
+          child: ExtentDelegateListView(
+            extentDelegate: FixedExtentDelegate(
+              computeLength: () => children.length,
+              computeExtent: (index) => children[index],
+            ),
+            childrenDelegate: SliverChildBuilderDelegate(
+              (context, index) => Text('${children[index]}'),
+              childCount: children.length,
+            ),
+          ),
+        ),
+      );
+
+      expect(controller.hasClients, isTrue);
+    });
+
+    testWidgets('inherits PrimaryScrollController explicitly', (tester) async {
+      final ScrollController controller = ScrollController();
+      await pumpList(
+        tester,
+        PrimaryScrollController(
+          controller: controller,
+          child: ExtentDelegateListView(
+            primary: true,
+            extentDelegate: FixedExtentDelegate(
+              computeLength: () => children.length,
+              computeExtent: (index) => children[index],
+            ),
+            childrenDelegate: SliverChildBuilderDelegate(
+              (context, index) => Text('${children[index]}'),
+              childCount: children.length,
+            ),
+          ),
+        ),
+      );
+
+      expect(controller.hasClients, isTrue);
+    });
+
+    testWidgets('does not inherit PrimaryScrollController - horizontal',
+        (tester) async {
+      final ScrollController controller = ScrollController();
+      await pumpList(
+        tester,
+        PrimaryScrollController(
+          controller: controller,
+          child: ExtentDelegateListView(
+            controller: ScrollController(),
+            scrollDirection: Axis.horizontal,
+            extentDelegate: FixedExtentDelegate(
+              computeLength: () => children.length,
+              computeExtent: (index) => children[index],
+            ),
+            childrenDelegate: SliverChildBuilderDelegate(
+              (context, index) => Text('${children[index]}'),
+              childCount: children.length,
+            ),
+          ),
+        ),
+      );
+
+      expect(controller.hasClients, isFalse);
+    });
+
+    testWidgets('does not inherit PrimaryScrollController - explicitly set',
+        (tester) async {
+      final ScrollController controller = ScrollController();
+      await pumpList(
+        tester,
+        PrimaryScrollController(
+          controller: controller,
+          child: ExtentDelegateListView(
+            primary: false,
+            controller: ScrollController(),
+            scrollDirection: Axis.horizontal,
```

Comment:
if `primary` was true, would `Axis.horizontal` here still prevent the scroll controller from being inherited?

Reply:
Nope. If `primary` is true, it will always inherit. I'll add a test to that effect.